miscmodule/showroll_command.go

```diff
@@ -0,0 +1,126 @@
+package miscmodule
+
+import (
+	"math"
+	"math/rand"
+	"regexp"
+	"strconv"
+	"strings"
+
+	bot "../sweetiebot"
+	"github.com/blackhole12/discordgo"
+)
+
+var diceregex = regexp.MustCompile("[0-9]*d[0-9]+")
+
+type showrollCommand struct {
+}
+
+func (c *showrollCommand) Info() *bot.CommandInfo {
+	return &bot.CommandInfo{
+		Name:              "Showroll",
+		Usage:             "Evaluates a dice expression.",
+		ServerIndependent: true,
+	}
+}
+func (c *showrollCommand) eval(args []string, index *int, info *bot.GuildInfo) string {
+	var s string
+	for *index < len(args) {
+		s += value(args, index) + "\n"
+	}
+	return s
+}
+func (c *showrollCommand) value(args []string, index *int, info *bot.GuildInfo) string {
+	*index++
+	if diceregex.MatchString(args[*index-1]) {
+		dice := strings.SplitN(args[*index-1], "d", 2)
+		var multiplier, num, threshold, fail int64 = 1, 1, 0, 0
+		var s string
+		s = "Rolling " + args[*index-1] + ": "
```

Comment:
You can collapse this into one line by doing `s := "Rolling " + args[*index-1] + ": "`